app/controllers/agentAppointment/AgentRelationshipController.scala

```diff
@@ -206,12 +213,41 @@ class AgentRelationshipController @Inject()(
   }
 
   def checkAnswers(): Action[AnyContent] = authenticated.andThen(withAppointAgentSession).async { implicit request =>
-    Future.successful(Ok(checkYourAnswers(request.sessionData)))
+    Future.successful(Ok(checkYourAnswers(submitAgentRelationship, request.sessionData)))
+  }
+
+  def submitAppointmentChanges(): Action[AnyContent] = authenticated.andThen(withAppointAgentSession).async {
+    implicit request =>
+      submitAgentRelationship.bindFromRequest.fold(
+        errors => {
+          Future.successful(
+            BadRequest(
+              checkYourAnswers(
+                errors,
+                request.sessionData
+              )))
+        }, { success =>
+          {
+            agentRelationshipService.sendAgentRelationshipRequest(
+              AgentRelationshipRequest(
+                agentRepresentativeCode = success.agentCode,
+                action = AppointmentAction.withName(success.action),
+                scope = AppointmentScope.withName(success.scope),
+                propertyLinks = None
+              )
+            )
+            Future.successful(Ok(confirmation(getAgentName(request))))
+          }
+        }
+      )
   }
 
   private def getAgentName(request: AppointAgentSessionRequest[AnyContent]) =
```

Comment:
this should be a generic function get Agent. which returns the agent details object.

Reply:
And if we require the agent address here, add a seperate refiner step on the methods that call this.

Reply:
i meant to change this. It was getAgentName when we only had the name. I started using Jon's agent details endpoint yesterday. Meant to refactor this.